src/serializationStrategies/XpathRangeSelector/xpath.ts

```diff
@@ -65,17 +69,26 @@ const resolveTextHighlightsToTextOffset = (element: Node, offset: number, contai
 };
 
 const floatThroughText = (element: Node, offset: number, container: Node): [Node, number] => {
-  if (isTextOrTextHighlight(element) && offset === 0 && element.parentNode && element.parentNode !== container) {
+  if (isTextOrTextHighlightOrScreenReaderNode(element) && offset === 0 && element.parentNode && element.parentNode !== container) {
     return floatThroughText(element.parentNode, nodeIndex(element.parentNode.childNodes, element), container);
-  } else if (isTextOrTextHighlight(element) && offset === getMaxOffset(element) && element.parentNode && element.parentNode !== container) {
+  } else if (isTextOrTextHighlightOrScreenReaderNode(element) && offset === getMaxOffset(element) && element.parentNode && element.parentNode !== container) {
+    return floatThroughText(element.parentNode, nodeIndex(element.parentNode.childNodes, element) + 1, container);
```

Comment:
what are all these +1s you're adding for?

Reply:
the diff is a little bit broken, I didn't touch this line
and about lines 76 - 83
- first check should be changed to `isTextOrTextHighlight` instead of `isTextOrTextHighlightOrScreenReaderNode` - I'll push a fix for this
- this `if branch` was added in order to fix test: `remove unnecessary text nodes with nested text highlights (on trailing edge)`
- from what I remember it was related to offset being less than max offset after passing through the first `floatThroughText` - in this case I had to check if `offset + 1` is equal to max offset and if the current offset is targeting screenreader node